### PR TITLE
pybind/mgr/balancer: more specific command outputs

### PR DESCRIPTION
**Pull request by xiexingguo**

### CODE REVIEW

File: src/pybind/mgr/balancer/module.py [hunk 1]
```diff
@@ -615,25 +620,35 @@ def optimize(self, plan):
...
+                detail = 'Please do "ceph balancer mode" to choose a valid mode first'
                 self.log.info('Idle')
+                return -errno.ENOEXEC, detail
             else:
-                self.log.info('Unrecognized mode %s' % plan.mode)
-        return False
-
+                detail = 'Unrecognized mode %s' % plan.mode
+                self.log.info(detail)
+                return -errno.EINVAL, detail
```
Comment (amitkumar50) [inline src/pybind/mgr/balancer/module.py]: NIT:tab

**Comment by xiexingguo** [comment]
@liewegas  Can you review this? Thanks!

**Comment by xiexingguo** [comment]
@liewegas Ping

File: src/pybind/mgr/balancer/module.py [hunk 2]
```diff
@@ -697,8 +697,13 @@ def do_crush_compat(self, plan):
         osdmap = ms.osdmap
         crush = osdmap.get_crush()
         pe = self.calc_eval(ms, plan.pools)
-        if pe.score == 0:
-            detail = 'Distribution is already perfect'
+        min_score_to_optimize = float(self.get_config('min_score', 0))
```
Comment (liewegas) [inline src/pybind/mgr/balancer/module.py]: should we make this .001 or something?

Comment (xiexingguo) [reply src/pybind/mgr/balancer/module.py]: I was thinking about that value too. But user can manually configure that anyway.

**Comment by xiexingguo** [comment]
http://pulpito.ceph.com/xxg-2018-02-08_01:34:34-rados-wip-xxg-testing-distro-basic-smithi/

**Pull request merged**